tests/tape/test_qnode.py

```diff
@@ -1016,6 +1018,114 @@ def circuit(a):
+            circuit.specs
+
+    def test_specs_backprop(self, qfunc):
+        """Tests specs property with backprop"""
+
+        dev = qml.device("default.qubit", wires=4)
+
+        circuit = qml.QNode(qfunc, dev, diff_method="backprop")
+
+        x = pnp.array([0.05, 0.1, 0.2, 0.3], requires_grad=True)
+        y = pnp.array(0.1, requires_grad=False)
+
+        res = circuit(x, y)
+
+        info = circuit.specs
+
+        assert len(info) == 9
+
+        assert info["by_size"] == defaultdict(int, {1: 2, 3: 1, 2: 1})
+        assert info["by_name"] == defaultdict(int, {"RX": 1, "Toffoli": 1, "CRY": 1, "Rot": 1})
+        assert info["total_operations"] == 4
+        assert info["total_observables"] == 2
+        assert info["num_tape_wires"] == 3
+        assert info["depth"] == 3
+        assert info["num_device_wires"] == 4
+        assert info["device_name"] == "default.qubit.autograd"
+        assert info["diff_method"] == "backprop"
+
+    def test_specs_parametershift(self, qfunc):
+        """Tests specs property with parameter-shift"""
+
+        dev = qml.device("default.qubit", wires=4)
+
+        circuit = qml.QNode(qfunc, dev, diff_method="parameter-shift")
+
+        x = pnp.array([0.05, 0.1, 0.2, 0.3], requires_grad=True)
+        y = pnp.array(0.1, requires_grad=False)
+
+        res = circuit(x, y)
+
+        info = circuit.specs
+
+        assert len(info) == 11
+
+        assert info["by_size"] == defaultdict(int, {1: 2, 3: 1, 2: 1})
+        assert info["by_name"] == defaultdict(int, {"RX": 1, "Toffoli": 1, "CRY": 1, "Rot": 1})
+        assert info["total_operations"] == 4
+        assert info["total_observables"] == 2
+        assert info["num_tape_wires"] == 3
+        assert info["depth"] == 3
+        assert info["num_trainable_params"] == 4
+        assert info["num_parameter_shift_executions"] == 7
+
+        assert info["num_device_wires"] == 4
+        assert info["device_name"] == "default.qubit"
+        assert info["diff_method"] == "parameter-shift"
+
+    def test_specs_adjoint(self, qfunc):
+        """Tests specs property with adjoint"""
+
+        dev = qml.device("default.qubit", wires=4)
+
+        circuit = qml.QNode(qfunc, dev, diff_method="adjoint")
+
+        x = pnp.array([0.05, 0.1, 0.2, 0.3], requires_grad=True)
+        y = pnp.array(0.1, requires_grad=False)
+
+        res = circuit(x, y)
+
+        info = circuit.specs
+
+        assert len(info) == 10
+
+        assert info["by_size"] == defaultdict(int, {1: 2, 3: 1, 2: 1})
+        assert info["by_name"] == defaultdict(int, {"RX": 1, "Toffoli": 1, "CRY": 1, "Rot": 1})
+        assert info["total_operations"] == 4
+        assert info["total_observables"] == 2
+        assert info["num_tape_wires"] == 3
+        assert info["depth"] == 3
+        assert info["num_trainable_params"] == 4
+
+        assert info["num_device_wires"] == 4
+        assert info["device_name"] == "default.qubit"
+        assert info["diff_method"] == "adjoint"

```

Comment:
I think it would be nice to add tests for:
* a qnode without gates
* a qnode where the tape changes with the arguments

Reply:
But nice to have the tests in different diff methods!

Reply:
Some slight variation across diff methods, like adding parameter shift executions or deleting `"num_trainbable_params"`.

Reply:
Would also be nice to have a case for the behaviour of having multiple measurements in the `return` statement of the qfunc.